### DISCUSSION TITLE
Activate extension if other extension is activated

### DISCUSSION THREAD

**Discussion by itavero** [microsoft/vscode-discussions#595] [Extension Development QnA] [answered]
I wrote an extension that depends on data generated from/provided by another extension (`ms-vscode.cmake-tools`).
Instead of duplicating the activation events from the `package.json` of `ms-vscode.cmake-tools`, is there a way to have my extension be activated automatically when `ms-vscode.cmake-tools` is activated?

I browsed through the documentation and the sources of vscode, but I wasn't able to find a solution for this.

**Comment by r3m0t** [comment]
If your extension is activated first you can use the extension api to activate the extension you depend on.

**Reply by itavero** [reply]
This is exactly not what I asked 😉

Ideally my only activation event would be that the other extension was activated.

**Comment by alefragnani** [comment, marked as answer]
Hi @itavero ,

There is not such event, as you can see here https://code.visualstudio.com/api/references/activation-events.

You can say your extension _depends on_ another extension, using the `extensionDependencies` entry in package.json (more details here https://code.visualstudio.com/api/references/extension-manifest), but this entry is for _installation purposes_, not activation. 

Right now, the only way is to _check_ if the `ms-vscode.cmake-tools` is active prior to run your commands/handle activation, using the the Extensions API (https://code.visualstudio.com/api/references/vscode-api#extensions). Something like:

```ts
    let msext = vscode.extensions.getExtension("ms-vscode.cmake-tools");
    if (!msext.isActive) return;
    ...
```

Otherwise, I suggest you to open a feature request in VS Code repo, so the team could take a look and evaluate the scenario you describe.

Hope this helps

**Comment by bwateratmsft** [comment]
Both of these require code in the primary extension (the one that activates first), but these are two options I've used in the past--
1. The primary extension registers a dummy command and calls it on itself. The secondary extension has an activation event for that dummy command.
2. The secondary extension has some form of custom marker put into its `contributes` property in `package.json`. The primary extension activates any other extension containing these markers.

I would love to see an issue filed on VSCode that would add a proper activation-on-another-extension-activation, so that code is _not_ needed in the primary extension.

**Comment by passeee** [comment]
Wouldn't this work with
`"activationEvents": [
    "onCommand:extension.sayHello"
]`?
